lib/model.js

```diff
@@ -3687,6 +3688,19 @@ Model.castObject = function castObject(obj, options) {
     }
 
     if (schemaType.$isMongooseDocumentArray) {
+      const castNonArraysOption = schemaType.options.castNonArrays != null ? schemaType.options.castNonArrays : schemaType.constructor.options.castNonArrays;
```

Comment:
Isn't this shorter and better (if castNonArrays can be null or undefined). 
```suggestion
      const castNonArraysOption = schemaType.options?.castNonArrays ??schemaType.constructor.options.castNonArrays;
```